tgui/packages/tgui/interfaces/Pandemic.tsx

```diff
@@ -0,0 +1,536 @@
+              <Button
+                key={resistance.name}
+                icon="eye-dropper"
+                disabled={!is_ready}
+                tooltip="Creates a vaccine bottle."
+                onClick={() =>
+                  act('create_vaccine_bottle', {
+                    index: resistance.id,
+                  })}>
+                {`${resistance.name}`}
+              </Button>
+            );
+          })}
+      </LabeledList.Item>
+    </LabeledList>
+  );
+};
+
+/** Displays info for the loaded blood, if any */
+const SpecimenDisplay = (_, context) => {
+  const { act, data } = useBackend<PandemicContext>(context);
+  const [tab, setTab] = useSharedState(context, 'tab', 0);
+  const { is_ready, viruses } = data;
+  if (!viruses?.length) {
+    return <NoticeBox>No viruses detected</NoticeBox>;
+  }
+  const virus = viruses[tab];
+  const setTabHandler = (index: number) => {
+    setTab(index);
+  };
+
+  return (
+    <Section
+      fill
+      scrollable
+      title="Specimen"
+      buttons={
+        <Stack>
+          {viruses.length > 1 && (
+            <Stack.Item>
+              <VirusTabs tab={tab} tabHandler={setTabHandler} />
+            </Stack.Item>
+          )}
+          <Stack.Item>
+            <Button
+              icon="flask"
+              content="Create culture bottle"
+              disabled={!is_ready}
+              onClick={() =>
+                act('create_culture_bottle', {
+                  index: virus.index,
+                })}
+            />
+          </Stack.Item>
+        </Stack>
+      }>
+      <Stack fill vertical>
+        <Stack.Item>
+          <VirusDisplay virus={virus} />
+        </Stack.Item>
+        <Stack.Item>
+          <SymptomDisplay symptoms={virus.symptoms} />
+        </Stack.Item>
+      </Stack>
+    </Section>
+  );
+};
+
+/** Virus Tab display - changes the tab for virus info
+ * Whenever the tab changes, the virus info is updated
+ */
+const VirusTabs = (props: TabsProps, context) => {
+  const { data } = useBackend<PandemicContext>(context);
+  const { tab, tabHandler } = props;
+  const { viruses } = data;
+  if (!viruses) {
+    return <NoticeBox>No viruses detected</NoticeBox>;
+  }
+
+  return (
+    <Tabs>
+      {viruses.map((virus, index) => {
+        return (
+          <Tabs.Tab
+            selected={tab === index}
+            onClick={() => tabHandler(index)}
+            key={virus.name}>
+            {virus.name}
+          </Tabs.Tab>
+        );
+      })}
+    </Tabs>
+  );
+};
+
+/** Displays info about the virus. Child elements display
+ * the virus's traits and descriptions.
+ */
+const VirusDisplay = (props: VirusDisplayProps) => {
+  const { virus } = props;
+
+  return (
+    <Stack fill>
+      <Stack.Item grow={3}>
+        <VirusTextInfo virus={virus} />
+      </Stack.Item>
+      {virus.is_adv && (
+        <>
+          <Stack.Divider />
+          <Stack.Item grow={1}>
+            <VirusTraitInfo virus={virus} />
```

Comment:
the one thing I don't like about this: using heavier componentization has lead to sending ```virus``` down twice, but there's no way  to fix this (that I'm seeing)